nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/service/canvas-context-menu.service.ts

```diff
@@ -477,14 +477,25 @@ export class CanvasContextMenu implements ContextMenuDefinitionProvider {
                 }
             },
             {
-                condition: (selection: any) => {
-                    // TODO - hasDetails
-                    return false;
+                condition: (selection: d3.Selection<any, any, any, any>) => {
+                    return this.canvasUtils.hasDetails(selection);
```

Comment:
The `hasDetails` logic needs updated to support a Process Group that a user does not have modify rights to. Currently the user can't "View Configuration" in this scenario. This is most likely just due to the fact that we separated Configure and Controller Services and the Configure option was the only option and visible for a read-only PG.

_Also_, once I updated the logic locally... I noticed that the order of the menu options was off. We show `Configure`, `Controller Services`. But when in read only mode, it is shown as `Controller Services`, `View Configuration`. We should move `View Configuration` before `Controller Services`.